File: day_00_template/python/input.py
file_path = "./day_00/input.txt"
test_data = []


def read_file_to_list(file_path):
    with open(file_path, "r") as file:
        return [line.strip() for line in file]


def get_input_data_as_string(test=False):
    if test:
        return "\n".join(test_data)
    with open(file_path, "r") as file:
        return file.read()


def get_input_data_as_list(test=False):
    if test:
        return test_data
    list_from_file = read_file_to_list(file_path)
    return list_from_file

File: day_00_template/go/input.go
package main

import (
    "bufio"
    "os"
    "strings"
)

var filePath = "./input.txt"
var testData = []string{

}

func ReadFileToList(filePath string) ([]string) {
    file, err := os.Open(filePath)
    if err != nil {
        return nil
    }
    defer file.Close()

    var lines []string
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        lines = append(lines, strings.TrimSpace(scanner.Text()))
    }
    return lines 
}

func GetInputDataAsString(test bool) (string) {
    if test {
        return strings.Join(testData, "\n")
    }

    data, err := os.ReadFile(filePath)
    if err != nil {
        return ""
    }
    return string(data)
}

func GetInputDataAsList(test bool) ([]string) {
    if test {
        return testData
    }
    return ReadFileToList(filePath)
}